spark/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaTableV2.scala

```diff
@@ -364,6 +364,16 @@ object DeltaTableV2 {
     try thunk catch {
       case e: InvalidProtocolVersionException if tableNameToUse.exists(_ != e.tableNameOrPath) =>
         throw e.copy(tableNameOrPath = tableNameToUse.get).initCause(e)
+      case e: DeltaTableFeatureException if Seq(
```

Comment:
nit: may be extend the `DeltaTableFeatureException` as `DeltaUnsupportedTableFeatureException` that has the defined arguments. You can avoid args number check and indexing specific arg.

Reply:
Done